[BUG] Regression in 0.8.0: Can’t use `#[event(start)]` without a direct `wasm-bindgen` dependency

### Is there an existing issue for this?

- [x] I have searched the existing issues

### What version of `workers-rs` are you using?

0.8.0

### What version of `wrangler` are you using?

n/a

### Describe the bug

After upgrading to v0.8, a start handler like this:

```rust
#[event(start)]
pub fn setup_hook() {
    // …
}
```

in a crate that does not have `wasm-bindgen` as a dependency, will fail to compile:

```
error[E0433]: cannot find module or crate `wasm_bindgen` in this scope
 --> lib.rs:LL:1
  |
LL | #[event(start)]
  | ^^^^^^^^^^^^^^^ use of unresolved module or unlinked crate `wasm_bindgen`
  |
  = note: this error originates in the attribute macro `event` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0433`.
error: could not compile `generate-newspaper-entry` (lib) due to 1 previous error
```

This is a consequence of [this change](https://github.com/cloudflare/workers-rs/commit/e363ba6#diff-5fbccf7aad8fecebdb79b59bcdf993ad2236c34b9360b7f6cc2862374497a610) where the import of `::worker::wasm_bindgen` was dropped.

### Steps To Reproduce

Cargo.toml
```toml
[package]
name = "issue-973"
version = "0.1.0"
edition = "2024"

[lib]
crate-type = ["cdylib"]

[dependencies]
# Only `worker` — no direct `wasm-bindgen` dependency.
# The `#[event(start)]` macro should re-export what it needs
# via `::worker::wasm_bindgen`, but since 0.8.0 it doesn't.
worker = "0.8.0"
```

src/lib.rs
```rust
use worker::event;

// Fails to compile with worker 0.8.0 because the `event` macro
// emits a bare `wasm_bindgen` reference instead of `::worker::wasm_bindgen`.
#[event(start)]
pub fn setup_hook() {}
```

--- a/.github/workflows/pullrequest.yml
+++ b/.github/workflows/pullrequest.yml
@@ -113,7 +113,7 @@ jobs:
 
       - name: Build all examples
         run: |
-          sed -i 's/, "examples\/axum"//g' Cargo.toml
+          sed -i '/"examples\/axum",/d' Cargo.toml
           for example in examples/*/; do
               example_name=$(basename "$example")
               if [ "$example_name" = "coredump" ]; then

--- a/examples/axum/Cargo.toml
+++ b/examples/axum/Cargo.toml
@@ -15,6 +15,6 @@ worker-macros = { workspace = true, features = ['http'] }
 axum = { version = "0.8", default-features = false, features = ['json'] }
 axum-macros = "0.5.0"
 tower-service = "0.3.3"
-wasm-bindgen-futures = "0.4"
-wasm-bindgen = "0.2.117"
+wasm-bindgen-futures = { workspace = true }
+wasm-bindgen = { workspace = true }
 serde = { version = "1.0", features = ["derive"] }

--- a/worker-macros/src/event.rs
+++ b/worker-macros/src/event.rs
@@ -230,11 +230,24 @@ pub fn expand_macro(attr: TokenStream, item: TokenStream) -> TokenStream {
         Start => {
             validate_event_fn(&input_fn, Start, 0, false);
 
+            let mod_name = Ident::new(
+                &format!("_worker_start_{}", input_fn.sig.ident),
+                input_fn.sig.ident.span(),
+            );
+
             let wasm_bindgen_code =
                 wasm_bindgen_macro_support::expand(quote! { start }, quote! { #input_fn })
                     .expect("wasm_bindgen macro failed to expand");
 
-            TokenStream::from(wasm_bindgen_code)
+            let output = quote! {
+                mod #mod_name {
+                    pub use ::worker::{wasm_bindgen, wasm_bindgen_futures};
+                }
+                use #mod_name::*;
+                #wasm_bindgen_code
+            };
+
+            TokenStream::from(output)
         }
     }
 }

--- a/worker/tests/event.rs
+++ b/worker/tests/event.rs
@@ -3,3 +3,9 @@ fn event_invalid_signatures() {
     let t = trybuild::TestCases::new();
     t.compile_fail("tests/ui/*.rs");
 }
+
+#[test]
+fn event_valid_signatures() {
+    let t = trybuild::TestCases::new();
+    t.pass("tests/pass/*.rs");
+}

--- a/worker/tests/pass/start-handler.rs
+++ b/worker/tests/pass/start-handler.rs
@@ -1,0 +1,14 @@
+// Regression test for https://github.com/cloudflare/workers-rs/issues/973.
+// A `#[event(start)]` handler must compile without `wasm-bindgen` as a direct
+// dependency. The macro should resolve `wasm_bindgen` through `::worker::`.
+// It must also not conflict when `wasm_bindgen` is already in scope, or when
+// multiple `#[event(start)]` handlers exist in the same module.
+use worker::{event, wasm_bindgen};
+
+#[event(start)]
+pub fn setup_hook() {}
+
+#[event(start)]
+pub fn another_hook() {}
+
+fn main() {}

